### PR TITLE
Feat/v0.76.3

### PR DESCRIPTION
**Pull request by nityanandagohain**
otel collector - 0.76.1
otel collector contrib - 0.76.3

See https://github.com/SigNoz/signoz-otel-collector/pull/128 for changes till 0.69.0

### CODE REVIEW

**Comment by nityanandagohain** [comment]
@makeavish  @srikanthccv  

I have done basic functionality testing, requesting you guys to check specifics of traces and metrics.

**Comment by nityanandagohain** [comment]
You will have to use go v1.19 for running this.

You can use this tool for running multiple versions of go https://github.com/kevincobain2000/gobrew

File: exporter/clickhouselogsexporter/exporter.go
Comment (srikanthccv) [inline exporter/clickhouselogsexporter/exporter.go:193]: > Important: Don't rely on this method to get a string identifier of TraceID. Use hex.EncodeToString explicitly instead. This method meant to implement Stringer interface for display purposes only.

https://pkg.go.dev/go.opentelemetry.io/collector/pdata/pcommon#TraceID.String

File: exporter/clickhousetracesexporter/clickhouse_exporter.go
Comment (srikanthccv) [inline exporter/clickhousetracesexporter/clickhouse_exporter.go:112]: Same

File: exporter/clickhousetracesexporter/clickhouse_exporter.go
Comment (makeavish) [inline exporter/clickhousetracesexporter/clickhouse_exporter.go:350]: @nityanandagohain We can use [traceUtil](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/2d65254ce3f42dd11934aa3f33ae77f242b0ed20/internal/coreinternal/traceutil/traceutil.go#L62) from contrib rather than creating our own utils.

Comment (nityanandagohain) [reply exporter/clickhousetracesexporter/clickhouse_exporter.go:350]: It's an internal package you won't be able to import it.

Comment (makeavish) [reply exporter/clickhousetracesexporter/clickhouse_exporter.go:350]: Got it

File: exporter/clickhouselogsexporter/config.go
Comment (srikanthccv) [inline exporter/clickhouselogsexporter/config.go:26]: remove instead of comment if it's not needed.

File: exporter/clickhouselogsexporter/exporter.go
Comment (srikanthccv) [inline exporter/clickhouselogsexporter/exporter.go:193]:
```suggestion
						utils.GetTraceId(r.TraceID()),
						utils.GetSpanId(r.SpanID()),
```
I would use more clear name, `ToHexString` or something.

File: processor/signozspanmetricsprocessor/config_test.go
Comment (srikanthccv) [inline processor/signozspanmetricsprocessor/config_test.go:102]:
```suggestion
```

File: processor/signozspanmetricsprocessor/processor_test.go
Comment (srikanthccv) [inline processor/signozspanmetricsprocessor/processor_test.go:113]:
```suggestion
					component.NewID(typeStr): tc.exporter,
```

**Review by srikanthccv — COMMENTED** [review]
Some minor comments.

```
go test -count=1 -v -race -cover ./processor/signozspanmetricsprocessor/...
go test -count=1 -v -race -cover ./processor/exporter/...
```

They both fail, I will add commits to fix them.

File: exporter/clickhouselogsexporter/exporter.go
Comment (nityanandagohain) [reply exporter/clickhouselogsexporter/exporter.go]: Done

**Review by makeavish — APPROVED** [review]

**Comment by nityanandagohain** [comment]
@srikanthccv how should we plan to release this, should we let it run on our staging env for few days?

**Comment by srikanthccv** [comment]
Yes, after this gets merged, the `main` will run on staging if a new query service deployment is pushed. The code changes as is look good to me. I am yet to look at the breaking changes for the migration path for the existing users, and I will approve after that.

**Review by srikanthccv — APPROVED** [review]
I have created an issue https://github.com/SigNoz/signoz-otel-collector/issues/130 for the migration path. I will update as I find more but this can get merged. However, all of them should be addressed before release.

**Comment by nityanandagohain** [comment]
Thanks @srikanthccv

**Pull request merged**
